File: models/shopping_list.py
from . import db
from datetime import datetime

class ShoppingList(db.Model):
    __tablename__ = 'shopping_lists'
    
    list_id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.user_id'), nullable=False)
    title = db.Column(db.String(255), nullable=False)
    status = db.Column(db.Enum('active', 'completed', 'archived'), default='active')
    total_items = db.Column(db.Integer, default=0)
    purchased_items = db.Column(db.Integer, default=0)
    estimated_total = db.Column(db.Numeric(10, 2), default=0.00)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    items = db.relationship('ShoppingListItem', backref='shopping_list', lazy='dynamic')
    
    def __repr__(self):
        return f'<ShoppingList {self.title}>'


class ShoppingListItem(db.Model):
    __tablename__ = 'shopping_list_items'
    
    item_id = db.Column(db.Integer, primary_key=True)
    list_id = db.Column(db.Integer, db.ForeignKey('shopping_lists.list_id'), nullable=False)
    name_vi = db.Column(db.String(200), nullable=False)
    name_en = db.Column(db.String(200))
    quantity = db.Column(db.Numeric(10, 2))
    unit = db.Column(db.String(20))
    category_code = db.Column(db.String(50))
    is_purchased = db.Column(db.Boolean, default=False)
    purchased_at = db.Column(db.DateTime)
    from_recipe_id = db.Column(db.Integer)
    from_recipe_title = db.Column(db.String(255))
    estimated_price = db.Column(db.Numeric(10, 2))
    actual_price = db.Column(db.Numeric(10, 2))
    notes = db.Column(db.Text)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    def __repr__(self):
        return f'<ShoppingListItem {self.name_vi}>'

File: dotnet_backend/BepTroLy.API/Models/ShoppingList.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BepTroLy.API.Models;

[Table("shopping_lists")]
public class ShoppingList
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("list_id")]
    public int ListId { get; set; }

    [Column("user_id")]
    public int UserId { get; set; }

    [Required]
    [MaxLength(255)]
    [Column("title")]
    public string Title { get; set; } = string.Empty;

    // Enum: 'active', 'completed', 'archived'
    [Column("status")]
    public string Status { get; set; } = "active";

    [Column("total_items")]
    public int TotalItems { get; set; } = 0;

    [Column("purchased_items")]
    public int PurchasedItems { get; set; } = 0;

    [Column("estimated_total", TypeName = "decimal(10, 2)")]
    public decimal EstimatedTotal { get; set; } = 0.00m;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    [ForeignKey("UserId")]
    public User? User { get; set; }

    public ICollection<ShoppingListItem> Items { get; set; } = new List<ShoppingListItem>();
}

[Table("shopping_list_items")]
public class ShoppingListItem
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("item_id")]
    public int ItemId { get; set; }

    [Column("list_id")]
    public int ListId { get; set; }

    [Required]
    [MaxLength(200)]
    [Column("name_vi")]
    public string NameVi { get; set; } = string.Empty;

    [MaxLength(200)]
    [Column("name_en")]
    public string? NameEn { get; set; }

    [Column("quantity", TypeName = "decimal(10, 2)")]
    public decimal? Quantity { get; set; }

    [MaxLength(20)]
    [Column("unit")]
    public string? Unit { get; set; }

    [MaxLength(50)]
    [Column("category_code")]
    public string? CategoryCode { get; set; }

    [Column("is_purchased")]
    public bool IsPurchased { get; set; } = false;

    [Column("purchased_at")]
    public DateTime? PurchasedAt { get; set; }

    [Column("from_recipe_id")]
    public int? FromRecipeId { get; set; }

    [MaxLength(255)]
    [Column("from_recipe_title")]
    public string? FromRecipeTitle { get; set; }

    [Column("estimated_price", TypeName = "decimal(10, 2)")]
    public decimal? EstimatedPrice { get; set; }

    [Column("actual_price", TypeName = "decimal(10, 2)")]
    public decimal? ActualPrice { get; set; }

    [Column("notes", TypeName = "text")]
    public string? Notes { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey("ListId")]
    public ShoppingList? ShoppingList { get; set; }
}
